cluster/src/main/scala/io/snappydata/gemxd/SparkSQLExecuteImpl.scala

```diff
@@ -483,7 +483,7 @@ object SnappyContextPerConnection {
     val context = connectionIdMap.get(connectionID)
     if (context != null) context
     else {
-      val snc = SnappyContext.getOrCreate(null: SparkContext).newSession()
+      val snc = SnappyContext(null: SparkContext).newSession()
```

Comment:
Spark 2.0  has very clear definition of SharedState and SessionState. SessionState is one which accepts a SQLConf (understandable) . But Spark 2.0 has one listener across all sessions. Its a bit weird, as different SQL conf might result in different query plans (e.g broadcast limit). How can a user make out , how his query got executed. May be its a Spark oversight.